frontend/src/metabase/collections/components/CollectionHeader/CollectionHeader.styled.tsx

```diff
@@ -37,6 +41,10 @@ export const CollectionHeaderButton = styled(
     color: ${color("brand")};
     background-color: ${color("bg-medium")};
   }
+
+  ${Button.Content} {
+    height: 100%;
+  }
```

Comment:
since we're shrinking the icon size, now the icon stays at the top, not center, so setting height to 100% fixes this since the button content is a flex container that center align its children.